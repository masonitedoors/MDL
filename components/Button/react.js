import React from 'react'
import classNames from 'classnames/bind'
import PropTypes from 'prop-types'
import { mCheck } from '@masonite/svg-icons'
import s from './style.module.scss'
import Icon from '../Icon/react'

const cx = classNames.bind(s)

const Button = ({
  children,
  className,
  disabled,
  fullWidth,
  onClick,
  variant,
  size,
  style,
  uppercase,
  active,
  ...props
}) => {
  const classes = cx('btn', {
    'btn--primary': variant === 'primary',
    'btn--secondary': ['secondary', 'dark'].includes(variant),
    'btn--light': variant === 'light',
    'btn--default': variant === 'default',
    'btn--uppercase': uppercase,
    'btn--sm': size === 'small' || size === 'sm',
    'btn--large': size === 'large',
    'btn--disabled': disabled,
    'btn--full-width': fullWidth,
    'btn--active': active,
    'btn--large-active': size === 'large' && active,
    'btn--small-active': (size === 'small' || size === 'sm') && active,
  }, className)

  return (
    <button
      type="button"
      className={classes}
      onClick={onClick}
      disabled={disabled}
      style={style}
      {...props}
    >
      {children}
      {active && size === 'large' && (
        <span className={cx('btn--active-check')}>
          <Icon svg={mCheck} size={16} />
        </span>
      )}
    </button>
  )
}

Button.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(['default', 'primary', 'secondary', 'light']),
  size: PropTypes.oneOf(['small', 'sm', 'large']),
  style: PropTypes.shape({}),
  uppercase: PropTypes.bool,
  fullWidth: PropTypes.bool,
}

Button.defaultProps = {
  active: false,
  className: null,
  disabled: false,
  onClick: null,
  variant: 'default',
  size: null,
  style: null,
  uppercase: false,
  fullWidth: false,
}

export default Button
