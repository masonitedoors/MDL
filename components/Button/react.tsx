import React, { FC, ReactNode } from 'react'
import classNames from 'classnames/bind'
import { mCheck } from '@masonite/svg-icons'
import s from './style.module.scss'
import Icon from '../Icon/react'

const cx = classNames.bind(s)

export interface ButtonProps {
  children: ReactNode
  className?: string
  disabled?: boolean
  fullWidth?: boolean
  onClick?: any
  variant?: 'default' | 'primary' | 'secondary' | 'light'
  size?: 'sm' | 'lg' | 'small' | 'large'
  style?: any
  uppercase?: boolean
  active?: boolean
  bold?: boolean
}

const Button: FC<ButtonProps> = ({
  children,
  className,
  disabled,
  fullWidth,
  onClick,
  variant,
  size,
  style,
  uppercase,
  active,
  bold,
  ...props
}) => {
  const classes = cx(
    'btn',
    {
      'btn--primary': variant === 'primary',
      'btn--secondary': ['secondary', 'dark'].includes(variant),
      'btn--light': variant === 'light',
      'btn--default': variant === 'default',
      'btn--uppercase': uppercase,
      'btn--sm': size === 'small' || size === 'sm',
      'btn--large': size === 'large',
      'btn--disabled': disabled,
      'btn--full-width': fullWidth,
      'btn--bold': bold,
      'btn--active': active,
      'btn--large-active': size === 'large' && active,
      'btn--small-active': (size === 'small' || size === 'sm') && active,
    },
    className,
  )

  return (
    <button
      type="button"
      className={classes}
      onClick={onClick}
      disabled={disabled}
      style={style}
      {...props}
    >
      {children}
      {active && size === 'large' && (
        <span className={cx('btn--active-check')}>
          <Icon svg={mCheck} size={16} />
        </span>
      )}
    </button>
  )
}

export default Button
