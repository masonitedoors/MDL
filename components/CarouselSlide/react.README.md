# Usage

Pass in children to the CarouselSlide to create a responsive carousel slider that works for all screenns and devices.
