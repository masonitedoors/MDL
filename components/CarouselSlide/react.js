import React from 'react';
// components
import ReactCarousel from 'react-elastic-carousel';
// icons
import { mChevronRight, mChevronLeft } from '@masonite/svg-icons';
// css
import classes from './style.module.scss';

const CarouselSlide = ({ children }) => {
  const breakPoints = [
    { width: 1, itemsToShow: 1, itemsToScroll: 1 },
    { width: 768, itemsToShow: 2, itemsToScroll: 2 },
    { width: 1200, itemsToShow: 3, itemsToScroll: 3 },
  ];

  return (
    <ReactCarousel breakPoints={breakPoints} showArrows={false} renderPagination={CarouselPagination}>
      {children}
    </ReactCarousel>
  );
};

const getPaginationOptions = (pages, activePage) => {
  if (pages.length === 0) {
    return [];
  }
  // return all page options if we have 5 or less
  if (pages.length <= 5) {
    return pages;
  }

  // if we are in the first 2 items, just return 5 options 0-4
  // since we know at this point we have more than 5 options
  if (activePage === 0 || activePage === 1) {
    return [0, 1, 2, 3, 4];
  }

  // if we are at one of the last 2 items then we need
  // to just return the last 5 pages
  if (activePage === pages.length - 1 || activePage === pages.length - 2) {
    return [pages.length - 5, pages.length - 4, pages.length - 3, pages.length - 2, pages.length - 1];
  }

  // otherwise we are somewhere in the middle, center the current page
  // and return 2 before and 2 after
  return [activePage - 2, activePage - 1, activePage, activePage + 1, activePage + 2];
};

const CarouselPagination = ({ pages = [], activePage, onClick }) => {
  return (
    <div className={classes.pagination}>
      <button disabled={activePage === 0} onClick={() => onClick(activePage - 1)}>
        <div
          dangerouslySetInnerHTML={{
            __html: mChevronLeft,
          }}
        />
      </button>
      {getPaginationOptions(pages, activePage).map((index) => (
        <button key={`pagination-index-${index}`} onClick={() => onClick(index)} className={activePage === index ? classes.active : ''}>
          {index + 1}
        </button>
      ))}
      <button disabled={activePage === pages.length - 1} onClick={() => onClick(activePage + 1)}>
        <div
          dangerouslySetInnerHTML={{
            __html: mChevronRight,
          }}
        />
      </button>
    </div>
  );
};

export default CarouselSlide;
