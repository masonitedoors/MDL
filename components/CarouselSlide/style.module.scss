@import 'styles/index';

.pagination {
  display: flex;
  margin-top: 2rem;
  justify-content: center;
  align-items: center;

  button {
    display: flex;
    justify-content: center;
    align-items: center;
    border: none;
    background-color: $color-white;
    color: $color-deepgrey;
    font-family: $font-family;
    font-size: 1.25rem;
    margin-left: 0.25rem;
    margin-right: 0.25rem;
    padding-left: 6px;
    padding-right: 6px;

    &:active,
    &:focus {
      border: none;
      outline: none;
    }

    &.active {
      font-weight: bold;
      color: $color-night;
    }

    &:hover {
      cursor: pointer;
    }

    &:first-of-type,
    &:last-of-type {
      margin: 0;

      svg {
        height: 42px;
        width: 42px;
      }
    }
  }
}
