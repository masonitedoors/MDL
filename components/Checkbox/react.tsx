import React, { FC } from 'react'
import { mCheck } from '@masonite/svg-icons'
import classNames from 'classnames/bind'
import style from './style.module.scss'

const cx = classNames.bind(style)

export interface CheckboxProps {
  checked: boolean
  disabled?: boolean
  onChange?: any
}

export const Checkbox: FC<CheckboxProps> = ({ checked, disabled = false, onChange = null }) => (
  <div
    role="checkbox"
    aria-checked={checked}
    tabIndex={0}
    className={cx('checkbox')}
    onClick={onChange}
    onKeyPress={onChange}
  >
    <input className={cx('checkbox-input')} type="checkbox" checked={checked} readOnly />
    <div className={cx('checkbox-svg-wrapper')}>
      <svg
        className={cx('checkbox-svg-icon', { 'checkbox-svg-icon--checked': checked })}
        dangerouslySetInnerHTML={{ __html: mCheck }}
      />
    </div>
  </div>
)

export default Checkbox
