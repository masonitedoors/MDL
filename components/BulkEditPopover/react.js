import React, { useEffect } from 'react'
import classNames from 'classnames/bind'
import { React as Checkbox } from 'components/Checkbox'
import { React as TextField } from 'components/TextField'
import styles from './style.module.scss'

const cx = classNames.bind(styles)

const BulkEditPopover = React.forwardRef(
  (
    {
      formControlType,
      formControlProps,
      left = 0,
      top = 0,
      width = 0,
      popDirection = 'top',
      visible = false,
      focused = false,
      onMouseLeave,
      clearAllCellStates,
      value,
    },
    ref,
  ) => {
    const bulkEditOutsideClickHandler = event => {
      if (ref.current) !ref.current.contains(event.target) && clearAllCellStates()
    }

    useEffect(() => {
      window.addEventListener('click', bulkEditOutsideClickHandler, true)
      return window.removeEventListener('click', bulkEditOutsideClickHandler)
    }, [])

    const getDirectionStyle = () => {
      const style = {
        top: {
          bottom: '100%',
          top: 'auto',
        },
        left: {
          right: '100%',
          left: 'auto',
        },
      }
      return style[popDirection]
    }

    switch (formControlType) {
      case TextField:
        formControlProps = {
          removeMargin: true,
          ...formControlProps,
        }
        break
      default:
    }

    return (
      <div
        ref={ref}
        className={cx('bulk-edit-popover', {
          'bulk-edit-popover--visible': visible || focused,
          'bulk-edit-popover--focused': focused,
          'bulk-edit-popover--w-textfield': formControlType === TextField,
          'bulk-edit-popover--w-checkbox': formControlType === Checkbox,
        })}
        style={{
          top,
          left,
          width,
          ...getDirectionStyle(),
        }}
        onMouseLeave={onMouseLeave}
        data-testid="bulkEditPopover"
      >
        {formControlType && React.createElement(formControlType, { ...formControlProps })}
      </div>
    )
  },
)

export default BulkEditPopover
