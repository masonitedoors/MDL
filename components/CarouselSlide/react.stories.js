import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import CarouselSlide from './react';
import TrendCard from '../TrendCard/react';
import Readme from './react.README.md';

storiesOf('CarouselSlide', module)
  .addParameters({ readme: { sidebar: Readme } })
  .addDecorator(withKnobs)
  .add('default', () => {
    const squareImage = boolean('Square image', false);
    return (
      <CarouselSlide>
        <TrendCard
          title='Why Stile and Rail Doors are Worth the Investment'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/img/masonite/8hui9fjtwo/exact/Harring_Hotel%20Indigo_2_L2.jpeg'
          squareImage={squareImage}
        />
        <TrendCard
          title='Attack Resistant Door Opening'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/download/masonite/ja05d7snlv/Attack-Resiatant_Library_mobil.jpg'
          squareImage={squareImage}
        />
        <TrendCard
          title='Reasons to Use Acoustically-Rated Doors in Healthcare Construction'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/img/masonite/v5g5khzopv/exact/CHUM_020706_10_N407.jpeg'
          squareImage={squareImage}
        />
        <TrendCard
          title='Adding Security Features to Wood Doors in Office Construction'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/download/masonite/rhcsfxmw8l/Etsy_CF041517_2.jpg'
          squareImage={squareImage}
        />
        <TrendCard
          title='The Most In-Demand Colors for Factory Painted Doors for 2020'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/img/masonite/yxzvpmqv3y/300x300px/MDF-3-Panel-MockShaker-CEN301-WinterWhite-bty.jpeg'
          squareImage={squareImage}
        />
        <TrendCard
          title='Variations of Authentic Wood Veneers for Doors'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/img/masonite/m6wbtcs5ft/exact/Hotel%20Indigo_bar%20area_L8.jpeg'
          squareImage={squareImage}
        />
        <TrendCard
          title='Why Stile and Rail Doors are Worth the Investment'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/img/masonite/8hui9fjtwo/exact/Harring_Hotel%20Indigo_2_L2.jpeg'
          squareImage={squareImage}
        />
        <TrendCard
          title='Attack Resistant Door Opening'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/download/masonite/ja05d7snlv/Attack-Resiatant_Library_mobil.jpg'
          squareImage={squareImage}
        />
        <TrendCard
          title='Reasons to Use Acoustically-Rated Doors in Healthcare Construction'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/img/masonite/v5g5khzopv/exact/CHUM_020706_10_N407.jpeg'
          squareImage={squareImage}
        />
      </CarouselSlide>
    );
  });
