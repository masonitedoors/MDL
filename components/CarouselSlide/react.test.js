import React from 'react';
import { render } from '@testing-library/react';
import CarouselSlide from './react';
import TrendCard from '../TrendCard/react';

describe('CarouselSlide', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(
      <CarouselSlide>
        <TrendCard
          title='Why Stile and Rail Doors are Worth the Investment'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/img/masonite/8hui9fjtwo/exact/Harring_Hotel%20Indigo_2_L2.jpeg'
          squareImage
        />
        <TrendCard
          title='Attack Resistant Door Opening'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/download/masonite/ja05d7snlv/Attack-Resiatant_Library_mobil.jpg'
          squareImage
        />
        <TrendCard
          title='Reasons to Use Acoustically-Rated Doors in Healthcare Construction'
          content='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
          image='https://embed.widencdn.net/img/masonite/v5g5khzopv/exact/CHUM_020706_10_N407.jpeg'
          squareImage
        />
      </CarouselSlide>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
