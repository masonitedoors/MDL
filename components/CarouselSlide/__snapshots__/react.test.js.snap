// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSlide should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-jSgupP ewJlEB rec rec-carousel-wrapper "
  >
    <div
      class="sc-eCssSg galrjd rec rec-carousel"
      style="height: 0px;"
    >
      <div
        class="sc-dlfnbm cukhLH rec rec-slider-container"
      >
        <div
          class="sc-hKgILt kFXEwu rec rec-slider"
          style="transition: all 500ms ease; left: 0px;"
        >
          <div
            class="rec rec-swipable"
            style="display: flex; flex-direction: row;"
          >
            <div
              class="rec rec-carousel-item rec-carousel-item-0 rec-carousel-item-visible"
            >
              <div
                class="sc-gsTCUz tXlBS rec rec-item-wrapper"
                id="0"
                style="width: 0px; padding: 0px 0px 0px 0px;"
              >
                <article
                  class="trend-card"
                >
                  <a>
                    <div
                      class="trend-card__content"
                    >
                      <h2
                        class="trend-card__title"
                      >
                        Why Stile and Rail Doors are Worth the Investment 
                        <div
                          class="trend-card__title-chevron"
                        >
                          <svg>
                            <svg
                              version="1.1"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <title>
                                chevron-right
                              </title>
                              <path
                                d="M15.7 11.3l-6-6c-0.4-0.4-1-0.4-1.4 0s-0.4 1 0 1.4l5.3 5.3-5.3 5.3c-0.4 0.4-0.4 1 0 1.4 0.2 0.2 0.4 0.3 0.7 0.3s0.5-0.1 0.7-0.3l6-6c0.4-0.4 0.4-1 0-1.4z"
                              />
                            </svg>
                          </svg>
                        </div>
                      </h2>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
                    </div>
                    <div
                      class="trend-card__image-wrapper"
                    >
                      <img
                        alt=""
                        class="trend-card__image"
                        src="https://embed.widencdn.net/img/masonite/8hui9fjtwo/exact/Harring_Hotel%20Indigo_2_L2.jpeg"
                      />
                    </div>
                  </a>
                </article>
              </div>
            </div>
            <div
              class="rec rec-carousel-item rec-carousel-item-1 rec-carousel-item-hidden rec-carousel-item-next"
            >
              <div
                class="sc-gsTCUz tXlBS rec rec-item-wrapper"
                id="1"
                style="width: 0px; padding: 0px 0px 0px 0px;"
              >
                <article
                  class="trend-card"
                >
                  <a>
                    <div
                      class="trend-card__content"
                    >
                      <h2
                        class="trend-card__title"
                      >
                        Attack Resistant Door Opening 
                        <div
                          class="trend-card__title-chevron"
                        >
                          <svg>
                            <svg
                              version="1.1"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <title>
                                chevron-right
                              </title>
                              <path
                                d="M15.7 11.3l-6-6c-0.4-0.4-1-0.4-1.4 0s-0.4 1 0 1.4l5.3 5.3-5.3 5.3c-0.4 0.4-0.4 1 0 1.4 0.2 0.2 0.4 0.3 0.7 0.3s0.5-0.1 0.7-0.3l6-6c0.4-0.4 0.4-1 0-1.4z"
                              />
                            </svg>
                          </svg>
                        </div>
                      </h2>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
                    </div>
                    <div
                      class="trend-card__image-wrapper"
                    >
                      <img
                        alt=""
                        class="trend-card__image"
                        src="https://embed.widencdn.net/download/masonite/ja05d7snlv/Attack-Resiatant_Library_mobil.jpg"
                      />
                    </div>
                  </a>
                </article>
              </div>
            </div>
            <div
              class="rec rec-carousel-item rec-carousel-item-2 rec-carousel-item-hidden"
            >
              <div
                class="sc-gsTCUz tXlBS rec rec-item-wrapper"
                id="2"
                style="width: 0px; padding: 0px 0px 0px 0px;"
              >
                <article
                  class="trend-card"
                >
                  <a>
                    <div
                      class="trend-card__content"
                    >
                      <h2
                        class="trend-card__title"
                      >
                        Reasons to Use Acoustically-Rated Doors in Healthcare Construction 
                        <div
                          class="trend-card__title-chevron"
                        >
                          <svg>
                            <svg
                              version="1.1"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <title>
                                chevron-right
                              </title>
                              <path
                                d="M15.7 11.3l-6-6c-0.4-0.4-1-0.4-1.4 0s-0.4 1 0 1.4l5.3 5.3-5.3 5.3c-0.4 0.4-0.4 1 0 1.4 0.2 0.2 0.4 0.3 0.7 0.3s0.5-0.1 0.7-0.3l6-6c0.4-0.4 0.4-1 0-1.4z"
                              />
                            </svg>
                          </svg>
                        </div>
                      </h2>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
                    </div>
                    <div
                      class="trend-card__image-wrapper"
                    >
                      <img
                        alt=""
                        class="trend-card__image"
                        src="https://embed.widencdn.net/img/masonite/v5g5khzopv/exact/CHUM_020706_10_N407.jpeg"
                      />
                    </div>
                  </a>
                </article>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pagination"
    >
      <button
        disabled=""
      >
        <div>
          <svg
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              chevron-left
            </title>
            <path
              d="M10.4 12l5.3-5.3c0.4-0.4 0.4-1 0-1.4s-1-0.4-1.4 0l-6 6c-0.4 0.4-0.4 1 0 1.4l6 6c0.2 0.2 0.5 0.3 0.7 0.3s0.5-0.1 0.7-0.3c0.4-0.4 0.4-1 0-1.4l-5.3-5.3z"
            />
          </svg>
        </div>
      </button>
      <button
        class="active"
      >
        1
      </button>
      <button
        class=""
      >
        2
      </button>
      <button
        class=""
      >
        3
      </button>
      <button>
        <div>
          <svg
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              chevron-right
            </title>
            <path
              d="M15.7 11.3l-6-6c-0.4-0.4-1-0.4-1.4 0s-0.4 1 0 1.4l5.3 5.3-5.3 5.3c-0.4 0.4-0.4 1 0 1.4 0.2 0.2 0.4 0.3 0.7 0.3s0.5-0.1 0.7-0.3l6-6c0.4-0.4 0.4-1 0-1.4z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
